### ISSUE TITLE
When will this be available?

### ISSUE THREAD

**Issue by trainerbill** [Intacct/intacct-sdk-js#1]
I am just starting an integration using node.  I only need invoicing at this time and was wondering if you had anything to commit so I can use and/or help build on.  Need to know ASAP.  Thanks!

**Comment by jimmymcpeter** [comment]
We do not have a firm delivery date yet.  If you need something immediately, I would look at GoodwayGroup's here -  
https://github.com/GoodwayGroup/intacct-api

**Comment by jimmymcpeter** [comment]
For anyone else looking at this, we link 3rd party libraries on our tools overview here - https://developer.intacct.com/tools/

**Comment by trainerbill** [comment]
@jimmymcpeter thanks for the quick response.  I figured as much since the repo was created 12 days ago.  I am using that third party module and its good, just not officially supported.  I don't want to be in the position of redoing my integration in a couple weeks if this library would be available.  I guess I was just wondering if you will be commiting as you go, so that I could maybe contribute the portions I need, once the basic design is done.

**Comment by jimmymcpeter** [comment]
It's probably going to be done in TypeScript and mirror the other SDK's.  The bulk of time right now is setting up the request handler and all of the logic in there.  How do you feel about promises?

**Comment by trainerbill** [comment]
Nice.  I love Typescript.  Promises are a must, probably make use of async/await in typescript.  So that was kinda my question.  I am assuming you will be using classes so if you make the base classes for the module and request handling, I would be willing to write some logic for the Invoicing portions I want.  That is assuming you have thought about the design :)

**Comment by trainerbill** [comment]
I would like to see a base class for request / response handling and then pass that class into the constructor of your API classes.  So I could create a Invoice class that takes the base class to handle request/response.  The API classes would extend some base module that has your functions for like readByQuery etc.

Ultimately I would do something like( assume IntacctClient is the request/response module)
```js
import { IntacctInvoice } from 'intacct';
let invoicing = new IntacctInvoice(IntacctClient);
invoicing.readByQuery(options).request();
```

I started something awhile back before I found the intacct-api module.  https://github.com/trainerbill/intacct.
